website/src/types/Chat.ts

```diff
@@ -124,6 +128,58 @@ export interface InferenceUpdateChatParams {
   hidden?: boolean;
 }
 
+export interface PluginEntry {
+  url: string;
+  enabled?: boolean;
+  trusted?: boolean;
+  spec?: object | null;
+  plugin_config?: PluginConfig | null;
+}
+
+export interface PluginApiType {
+  type: string;
+  url: string;
+  has_user_authentication: boolean | null;
+  // NOTE: Some plugins using this field,
+  // instead of has_user_authentication
+  is_user_authenticated: boolean | null;
```

Comment:
☹

Reply:
?

Reply:
why does it have to be like this? is it something external that we cannot control?

Reply:
Yes there are some plugins available today, that have not respected OpenAI spec, and that is why we have two fields. correct one is has_user_authentication.